extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracePropagator.java

```diff
@@ -108,14 +109,16 @@ public <C> Context extract(Context context, @Nullable C carrier, TextMapGetter<C
     if (carrier != null) {
       BaggageBuilder baggageBuilder = Baggage.builder();
       for (String key : getter.keys(carrier)) {
-        if (!key.startsWith(PREFIX_BAGGAGE_HEADER)) {
+        String lowercaseKey = key.toLowerCase(Locale.ROOT);
+        if (!lowercaseKey.startsWith(PREFIX_BAGGAGE_HEADER)) {
           continue;
         }
         String value = getter.get(carrier, key);
         if (value == null) {
           continue;
         }
-        baggageBuilder.put(key.replace(OtTracePropagator.PREFIX_BAGGAGE_HEADER, ""), value);
+        String baggageKey = lowercaseKey.replace(OtTracePropagator.PREFIX_BAGGAGE_HEADER, "");
+        baggageBuilder.put(baggageKey, value);
```

Comment:
```suggestion
        baggageBuilder.put(key.substring(OtTracePropagator.PREFIX_BAGGAGE_HEADER.length()), value);
```
Using substring preserves the case of the baggage key

Reply:
We can not preserve the case of the baggage key anyway. They are carried in http headers and headers are case insensitive. The servlet implementations, and upstream services (sometimes) change it. I have seen several type of changes:

- `ot-baggage-some-key` to `Ot-Baggage-Some-Key` (for example golang http package does this)
- `ot-baggage-Some-Key` to `ot-baggage-some-key` (tomcat does this)
- `ot-baggage-Some-Key` to `ot-baggage-Some-Key` (keeping the case, undertow does this)

I think, since there is no guarantee of baggage name casing, we can at least keep it consistent and lowercase it everywhere (as tomcat does). Of course I have no strong objection with your changes and if you still prefer your changes I can apply them.

Reply:
Nit: I think it would be better to use substring on the `lowercaseKey`. Substring is a simpler operation than replace and should be a bit more efficient. Also replace replaces all occurrences of given substring which would have undesired results if the key contained multiple copies of the prefix (which I guess is unlikely to ever happen).

Reply:
Done!